### PR TITLE
Extended messages

### PR DESCRIPTION
**Pull request by CallumCoy**
Ta plugin, code plugin, and url handler can now send extended messages without issues.

### CODE REVIEW

**Review by c650 — COMMENTED** [review]
what does this do

File: src/app/plugins/code.plugin.ts [hunk 1]
```diff
@@ -30,31 +29,25 @@ export class CodePlugin extends Plugin {
...
-        .catch((err) => {
-          const channelName = this.container.messageService.getChannel(message).name;
-          this.container.loggerService.warn(
-            `Failed to find message id: ${messageID} in the following channel ${channelName}, extra information if any:\n${err}`
-          );
-        });
-      return;
+    if (!messageID) {
+      message.channel.send(this.formattingMessage);
     }
```
Comment (tanndlin) [inline src/app/plugins/code.plugin.ts:34]: return after this

File: src/app/plugins/code.plugin.ts [hunk 2]
```diff
@@ -30,31 +29,25 @@ export class CodePlugin extends Plugin {
...
+      .then((targMessage) => {
+        const messageToSend = `\`\`\`${language}\n ${targMessage.content}\n\`\`\``;
+        message.channel.send(messageToSend, {
+          split: { prepend: `\`\`\`${language}\n`, append: `\`\`\`` },
+        });
+      })
+      .catch((err) => {
+        const channelName = this.container.messageService.getChannel(message).name;
+        this.container.loggerService.warn(
+          `Failed to find message id: ${messageID} in the following channel ${channelName}, extra information if any:\n${err}`
```
Comment (tanndlin) [inline src/app/plugins/code.plugin.ts:47]: why not just pass in `message.channel.name`?

File: src/app/plugins/ta.plugin.ts [hunk 3]
```diff
@@ -71,7 +71,9 @@ export class TaPlugin extends Plugin {
       .array()
       .map((ta) => ta.user.toString()) //Convert to pingable mentions
       .join(' ');
-    message.channel.send(`${message.author} asks: \n> ${question}\n${mentions}`);
+    message.channel.send(`${message.author} asks: \n>>> ${question}\n${mentions}`, {
+      split: { prepend: '>>>' },
+    });
```
Comment (tanndlin) [inline src/app/plugins/ta.plugin.ts:76]: Did you test this?

**Comment by CallumCoy** [comment]
> what does this do

basically added message options "Ta plugin, code plugin, and url handler" so they can send messages over 2000 characters

File: src/app/plugins/ta.plugin.ts (hunk 3 above)
Comment (CallumCoy) [reply src/app/plugins/ta.plugin.ts:76]: All have been tested

Comment (tanndlin) [reply src/app/plugins/ta.plugin.ts:76]: Pretty sure this puts the mentions in a quote too

Comment (CallumCoy) [reply src/app/plugins/ta.plugin.ts:76]: It does, but it looks worse if the message comes out of the quotes than with the mention inside the quotes.

Comment (CallumCoy) [reply src/app/plugins/ta.plugin.ts:76]: I'll post images of results in discord in a second.

File: src/app/plugins/code.plugin.ts (hunk 2 above)
Comment (CallumCoy) [reply src/app/plugins/code.plugin.ts:47]: doesn't seem to be an element

**Review by tanndlin — APPROVED** [review]

**Pull request merged**